### ISSUE TITLE
"SyntaxError: Unexpected token '.'" when uglifying 

### ISSUE THREAD

**Issue by IlyaRepo** [facebook/react-native#8290]
Whenever I turn off debugging from chrome I get red screen with

"SyntaxError: Unexpected token '.'"

As soon as I turn on chrome debugger all errors go away. I tried building the bundle by running "react-native bundle --platfrom ios --dev false --entry-file index.ios.js --bundle-output main.jsbundle" and got the same issue. From the logs I can see the following:

Worker Farm: Received message for unknown index for existing child. This should not happen! Tue, 21 Jun 2016 22:17:13 GMT ReactNativePackager:SocketServer request error { message: 'Unexpected token: punc (.)', filename: 0, line: 19, col: 6, pos: 964, stack: 'Error\n at new JS_Parse_Error (eval at (/Users/aaa/bbb/src/ccc/node_modules/uglify-js/tools/node.js:22:1), .......

I cannot figure out which '.' exactly the packager is complaining and how Chrome debugging makes the error go away.

 "react": "^0.14.5", "react-native": "^0.24.1", only tried iOS on Mac

**Comment by acresta88** [comment]
Why is this issue closed? I'm getting the same bug with RN 0.40 is there a way to fix it? can you share your solution if you have one. Thanks

**Comment by CharlesMangwa** [comment]
**TL;DR:** This kind of issue come from an error you made in your code by writing the specified character in the error message. For me it was the use of the new exponentiation operator `**`.

______

I've faced a similar error and I too, didn't find the answer online. So I though I might just share my solution here if it can help anybody else.
Here is what my error looked like: 

<img src="https://user-images.githubusercontent.com/8975443/28233850-33d45948-68fa-11e7-84d8-e7de0bf60b0e.png" width="275" alt="SyntaxError: Unexpected token '*'" />

Nothing really helpful as you can see. So I debugged my app with the old comment/uncomment method, component after component. And I finally managed to identify the failing piece of code: 

```javascript
1  _formatDistance = (distance: number): string => {
2   let result
3   if (distance > 10 ** 3) {
4      result = `${(distance.toFixed(2) / 1000)
5        .toFixed(2)
6        .toString()
7        .replace('.', ',')} KM`
8    }
9    else result = `${distance} M`
10   return result
11 }
```

As you can see line `3`, I was using the new exponentiation operator ([Stage 4 ECMA's proposal](https://github.com/rwaldron/exponentiation-operator)) but for whatever reason, React Native doesn't want it, even with the good Babel presets. 
So to fix my issue, I just replace this line by:
```javascript
if (distance > 1000) {
```
And I was back on track!

**Comment by l540144279** [comment]
@acresta88 I have met the same Error! Why is this issue closed without solution?！Do yon have a solution? please share,  Thank you very much。RN 0.46.4
